llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -18594,7 +18608,10 @@ static SDValue performBuildShuffleExtendCombine(SDValue BV, SelectionDAG &DAG) {
                                    : BV.getOperand(1).getOperand(0),
                                cast<ShuffleVectorSDNode>(BV)->getMask());
   }
-  return DAG.getNode(IsSExt ? ISD::SIGN_EXTEND : ISD::ZERO_EXTEND, DL, VT, NBV);
+  unsigned ExtOpc = !SeenZExtOrSExt ? ISD::ANY_EXTEND
+                    : IsSExt        ? ISD::SIGN_EXTEND
+                                    : ISD::ZERO_EXTEND;
```

Comment:
Minor nitpick: I'm not keen on the readability of nested ternary operators like this. Can we add some brackets to make it clear
```suggestion
  unsigned ExtOpc = !SeenZExtOrSExt ? ISD::ANY_EXTEND
                    :(IsSExt        ? ISD::SIGN_EXTEND
                                    : ISD::ZERO_EXTEND);
```

Reply:
Ah yep good point - have added the brackets to improve readability (plus the re-formatting from clang-format).

Thank you for reviewing!